src/main/scala/org/apache/spark/sql/hive/llap/LlapExternalCatalog.scala

```diff
@@ -91,6 +92,32 @@ private[spark] class LlapExternalCatalog(
     }
   }
 
+  override def createTable(
+      tableDefinition: CatalogTable,
+      ignoreIfExists: Boolean): Unit = {
+    logInfo(tableDefinition.toString)
+
+    assert(tableDefinition.identifier.database.isDefined)
+    val db = tableDefinition.identifier.database.get
+    requireDbExists(db)
+
+    if (tableExists(db, tableDefinition.identifier.table)) {
+      if (ignoreIfExists) {
+        // No-op
+      } else {
+        throw new TableAlreadyExistsException(db = db, table = tableDefinition.identifier.table)
+      }
+    } else {
+      val sessionState =
+        SparkSession.getActiveSession.get.sessionState.asInstanceOf[LlapSessionState]
+      val stmt = sessionState.connection.createStatement()
+      // Check the privilege by creating a dummy table with the given name.
+      stmt.executeUpdate(s"CREATE TABLE ${tableDefinition.identifier.quotedString} (dummy INT)")
+      super.dropTable(db, tableDefinition.identifier.table, true, true)
```

Comment:
Note that `super.dropTable` can drop table without any access privilege because it's Spark-native.